app/services/package.scala

```diff
@@ -0,0 +1,59 @@
+import com.gu.identity.play.IdMinimalUser
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+package object services {
+
+  case class UserId(id:String) {
+    override def toString = id
+  }
+
+  object UserId {
+    val paramName = "user-id"
+
+    implicit val jsonReader = new Reads[UserId] {
+      override def reads(json: JsValue): JsResult[UserId] = json match {
+        case JsString(id) => JsSuccess(UserId(id))
+        case _            => JsError("Cannot parse UserId")
+      }
+    }
+  }
+
+  case class IdentityToken(token: String) {
+    override def toString = token
+  }
+
+  object IdentityToken {
+    val paramName = "identity-token"
+
+    implicit val jsonReader = new Reads[IdentityToken] {
+      override def reads(json: JsValue): JsResult[IdentityToken] = json match {
+        case JsString(id) => JsSuccess(IdentityToken(id))
+        case _            => JsError("Cannot parse IdentityToken")
+      }
+    }
+  }
+
+  sealed trait UserIdData {
+    def id: UserId
+  }
+
+  case class MinimalIdUser(user: IdMinimalUser) extends UserIdData {
```

Comment:
The idea here is to define an [algebraic data type](http://twitter.github.io/effectivescala/#Functional programming-Case classes as algebraic data types) to represent both authenticated and guest users (both having an `id` method).